Automattic/wp-calypso
# Media: Refactor media storage to use `react-query`

#### Changes proposed in this Pull Request

Currently, the media storage tree is a subtree of the sites tree. Because of that, it's always loaded with the monolithic reducer, as the sites tree is always loaded. However, the media storage state is used only for simple fetching, plus it has no relation to the rest of the redux state tree whatsoever. This makes it a perfect candidate for refactoring to `react-query`, and that's confirmed by the amount of boilerplate we're removing (+280 -828 LOC).

#### Testing instructions

* Go to `/devdocs/blocks/plan-storage` and verify it looks/works the same as before.
* Go to `/export/:site` where `:site` is a WP.com simple site.
* Verify that if the site has no media, you don't see the media export card.
* Verify that if the site has media, you see the media export card.
* Go to `/settings/performance/:site` on a Jetpack or Atomic site with a high-tier plan
* Verify the video hosting is enabled in the Media card, and the usage still looks/works as it does in production.
* Verify uploading a media file from the media modal and from `/media/:site` still works well.
* Verify deleting a media file from the media modal and from `/media/:site` still works well.
* Verify all tests pass.
